### PR TITLE
ATDM: ats1: Final cleanup (hopefully) (ATDV-361, #7470)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
* Disable ROL tests in `Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt` build (see #7470)
* Don't install the intel-18 builds (SPARC does not support intel-18 on this platform and EMPIRE does not use these installs)  (Saving disk space)

See [ATDV-361](https://sems-atlassian-srn.sandia.gov/browse/ATDV-361).

## How was this tested

On 'mutrino' I ran:

```
$ ssh mutirno

$ cd /lscratch1/rabartl/Trilinos.base/BUILDS/ATS1/CTEST_S/

$ rm -r Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt

$ env \
    SBATCH_TEST_TIME_LIMIT_MINUTES=1200 \
  ./ctest-s-local-test-driver.sh \
    ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt

***
*** ./ctest-s-local-test-driver.sh
***

ATDM_TRILINOS_DIR = '/lscratch1/rabartl/Trilinos.base/Trilinos'

Load some env to get python, cmake, etc ...

Hostname 'mutrino' matches known ATDM host 'mutrino' and system 'ats1'
Setting compiler and build options for build-name 'default'
Using ats1 compiler stack INTEL-18.0.5_MPICH-7.7.6 to build DEBUG code with Kokkos node type SERIAL and KOKKOS_ARCH=HSW

Running builds:
    ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt

Thu Jun  4 12:50:02 MDT 2020

Running Jenkins driver Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt.sh ...

    Creating directory: Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt/smart-jenkins-driver.out

real    397m30.593s
user    0m1.969s
sys     0m9.114s

99% tests passed, 11 tests failed out of 2084

Thu Jun  4 19:27:32 MDT 2020

Done running all of the builds!
```

And that submitted to CDash at:

* https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt-exp&field2=buildstamp&compare2=61&value2=20200604-1850-Experimental

showing:

| Site | Build Name | C Err | C Warn | C Time | B Err | B Warn | B Time | Not Run | Fail | Pass | Time | Proc Time | Start Time | Labels |
| -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- |
mutrino | Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt-exp | 0 | 4 | 11m 51s | 0 | 50 | 3h 3m 9s | 0 | 11 | 2073 | 3h 17m 20s | 7h 19m 39s | 6 hours ago | (28 labels)

So the tests finished in 3h 17m minus ROL tests and no build errors!

The full details are shown in [ATDV-361](https://sems-atlassian-srn.sandia.gov/browse/ATDV-361?focusedCommentId=60315&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-60315).

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
CC @e10harvey 

And bingo, we have full test results for all of the new 'ats1' builds today as shown [here](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2020-06-05&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-ats1&field2=buildname&compare2=64&value2=-exp) which includes test results for the build

* `Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt`

showing the full set of builds:


| Site | Build Name | Update | Update Time | C Err | C Warn | C Time | B Err | B Warn | B Time | Not Run | Fail | Pass | Time | Proc Time | Start Time | Labels |
| -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- |
mutrino | Trilinos-atdm-ats1-hsw_intel-18.0.5_mpich-7.7.6_openmp_static_dbg | 7a5eb7 | 1m 36s | 0 | 4 | 8m 47s | 0 | 0 | 17s | 0 | 0 | 0 | 0s | 0s | 5 hours ago | (28 labels)
mutrino | Trilinos-atdm-ats1-hsw_intel-18.0.5_mpich-7.7.6_openmp_static_opt | 7a5eb7 | 1m 36s | 0 | 4 | 16m 21s | 0 | 4 | 1m 53s | 0 | 11 | 2255 | 3h 51m 2s | 8h 28m 17s | 6 hours ago | (28 labels)
mutrino | Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_dbg | 7a5eb7 | 1m 30s | 0 | 4 | 8m 33s | 0 | 0 | 58m 13s | 0 | 0 | 0 | 0s | 0s | 6 hours ago | (28 labels)
mutrino | Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt | 7a5eb7 | 2m 6s | 0 | 4 | 15m 21s | 0 | 4 | 13m 37s | 0 | 11 | 2076 | 3h 17m 58s | 7h 19m 57s | 6 hours ago | (28 labels)
mutrino | Trilinos-atdm-ats1-knl_intel-18.0.5_mpich-7.7.6_openmp_static_dbg | 7a5eb7 | 1m 24s | 0 | 4 | 8m 59s | 0 | 0 | 17s | 0 | 0 | 0 | 0s | 0s | 6 hours ago | (28 labels)
mutrino | Trilinos-atdm-ats1-knl_intel-18.0.5_mpich-7.7.6_openmp_static_opt | 7a5eb7 | 1m 30s | 0 | 4 | 15m 48s | 0 | 4 | 2m 21s | 0 | 18 | 2243 | 5h 55m 23s | 12h 15m 13s | 7 hours ago | (28 labels)
mutrino | Trilinos-atdm-ats1-knl_intel-19.0.4_mpich-7.7.6_openmp_static_dbg | 7a5eb7 | 1m 48s | 0 | 4 | 9m 10s | 0 | 0 | 43m 56s | 0 | 0 | 0 | 0s | 0s | 6 hours ago | (28 labels)
mutrino | Trilinos-atdm-ats1-knl_intel-19.0.4_mpich-7.7.6_openmp_static_opt | 7a5eb7 | 1m 48s | 0 | 4 | 14m 46s | 0 | 4 | 12m 19s | 0 | 18 | 2243 | 5h 50m 8s | 12h 1m 3s | 7 hours ago | (28 labels)

and we are only getting installs for the 'intel-19' builds, not the 'intel-18' builds as shown by:

```
$ ssh mutrino

$ ls -l /projects/atdm_devops/trilinos_installs/2020-06-05/ | grep ats1-
drwxrwxr-x 6 jenkins wg-run-as-atdm-devops 4096 Jun  5 00:50 ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_dbg
drwxrwxr-x 7 jenkins wg-run-as-atdm-devops 4096 Jun  5 00:28 ats1-hsw_intel-19.0.4_mpich-7.7.6_openmp_static_opt
drwxrwxr-x 6 jenkins wg-run-as-atdm-devops 4096 Jun  5 00:29 ats1-knl_intel-19.0.4_mpich-7.7.6_openmp_static_dbg
drwxrwxr-x 7 jenkins wg-run-as-atdm-devops 4096 Jun  5 00:12 ats1-knl_intel-19.0.4_mpich-7.7.6_openmp_static_opt
```

So it looks like this PR is good to go.

File: cmake/std/atdm/README.md [hunk 1]
```diff
@@ -1654,14 +1654,22 @@ ENDIF()
...
-(values `CUDA`, `OPENMP`, `SERIAL`), `ATDM_CUDA_RDC` (`ON`/`OFF`), `ATDM_FPIC`
-(`ON`/`OFF`), `ATDM_COMPLEX` (`ON`/`OFF`), `ATDM_SHARED_LIBS` (`ON`/`OFF`),
-`ATDM_CMAKE_BUILD_TYPE` (values `DEBUG`, `RELEASE`, and `RELEASE-DEBUG`),
-`Trilinos_ENABLE_DEBUG` (`ON`/`OFF`), and `ATDM_PT_PACKAGES` (`ON`/`OFF`).  No
-other variables should be used in if-logic in these files as other variables
-may change in the future.
+recommended variables are:
+
+* `ATDM_COMPILER` (uppercase)
+* `ATDM_KOKKOS_ARCH_JOB_NAME_KEYS` (uppercase and seprated by `_`)
```
Comment (jmgate) [inline cmake/std/atdm/README.md:1660]: typo

File: cmake/std/atdm/README.md [hunk 2]
```diff
@@ -1654,14 +1654,22 @@ ENDIF()
...
+* `ATDM_NODE_TYPE` (values `CUDA`, `OPENMP`, `SERIAL`)
+* `ATDM_KOKKOS_ARCH` (uppercase and seprated by `,`)
```
Comment (jmgate) [inline cmake/std/atdm/README.md:1662]: typo

File: cmake/std/atdm/README.md (hunk 1 above)
Comment (bartlettroscoe) [reply cmake/std/atdm/README.md]: Addressed in commit 7d79c83.

File: cmake/std/atdm/README.md (hunk 2 above)
Comment (bartlettroscoe) [reply cmake/std/atdm/README.md]: Addressed in commit 7d79c83

**Comment by bartlettroscoe** [comment]
Thanks @jmgate!

**Pull request merged**